File: playground/devfancy/commerce2/src/main/java/team/sipe/commerce/order/presentation/OrderController.java
package team.sipe.commerce.order.presentation;

import team.sipe.commerce.order.application.dto.OrderDetailsResponse;
import team.sipe.commerce.order.application.OrderService;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class OrderController {

    private final OrderService orderService;

    public OrderController(final OrderService orderService) {
        this.orderService = orderService;
    }

    @GetMapping(path = "/api/orders/{id}")
    public ResponseEntity<OrderDetailsResponse> findByOrderDetail(@PathVariable final Long id) {
        OrderDetailsResponse response = orderService.findOrderDetails(id);
        return new ResponseEntity<>(response, HttpStatus.OK);
    }


    @GetMapping(path = "/api/orders/mongodb/{id}")
    public ResponseEntity<OrderDetailsResponse> findByOrderDetailWithMongoDb(@PathVariable final Long id) {
        OrderDetailsResponse response = orderService.findOrderDetailsWithSnapshot(id);
        return new ResponseEntity<>(response, HttpStatus.OK);
    }
}

File: playground/ruthetum/app/e-commerce/order/src/main/kotlin/team/sipe/spurt/ecommerce/order/controller/OrderController.kt
package team.sipe.spurt.ecommerce.order.controller

import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RestController
import team.sipe.spurt.ecommerce.order.dto.response.OrderDetailsResponse
import team.sipe.spurt.ecommerce.order.service.OrderService

@RestController
class OrderController(
    private val orderService: OrderService,
) {
    @GetMapping("/v1/orders/{id}")
    fun v1(
        @PathVariable id: Long,
    ): ResponseEntity<OrderDetailsResponse> {
        return ResponseEntity.ok(
            OrderDetailsResponse(
                orderService.get(id)
            )
        )
    }
}